src/vmm/src/devices/virtio/rng/entropy_metrics.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the metrics system for entropy devices.
+//!
+//! # Metrics format
+//! The metrics are flushed in JSON when requested by vmm::logger::metrics::METRICS.write().
+//!
+//! ## JSON example with metrics:
+//! ```json
+//!  "entropy": {
+//!     "activate_fails": "SharedIncMetric",
+//!     "entropy_event_fails": "SharedIncMetric",
+//!     "entropy_event_count": "SharedIncMetric",
+//!     ...
+//!  }
+//! }
+//! ```
+//! Each `entropy` field in the example above is a serializable `EntropyDeviceMetrics` structure
+//! collecting metrics such as `activate_fails`, `entropy_event_fails` etc. for the entropy device.
+//! Since entropy doesn't support multiple devices, there is no per device metrics and
+//! `entropy` represents the aggregate entropy metrics.
+//!
+//! # Design
+//! The main design goals of this system are:
+//! * Have a consistent approach of keeping device related metrics in the individual devices
+//!   modules.
+//! * To decouple entropy device metrics from logger module by moving EntropyDeviceMetrics out of
+//!   FirecrackerDeviceMetrics.
+//! * Rely on `serde` to provide the actual serialization for writing the metrics.
+//!
+//! The system implements 1 type of metrics:
+//! * Shared Incremental Metrics (SharedIncMetrics) - dedicated for the metrics which need a counter
+//! (i.e the number of times an API request failed). These metrics are reset upon flush.
+
+use serde::ser::SerializeMap;
+use serde::{Serialize, Serializer};
+
+use crate::logger::SharedIncMetric;
+
+/// Stores aggregated entropy metrics
+pub(super) static ENTROPY_METRICS: EntropyDeviceMetrics = EntropyDeviceMetrics::new();
```

Comment:
nit: I would rather if we just called this `METRICS`. We do not need to specify the name here. If we ever need to differentiate between two similarly named structs we should use the namespacing providing by rust modules, e.g. `devices::virtio::entropy::METRICS`